### PR TITLE
feat(APIv2): RHINENG-12395 provide timestamp of last scan for system

### PR DESCRIPTION
**Pull request by romanblanco**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: spec/controllers/v2/test_results_controller_spec.rb [hunk 1]
```diff
@@ -114,6 +115,33 @@
           expect(response_body_data).to match_array(items.map { |item| a_hash_including('id' => item.id) })
         end
       end
+
+      context 'with multiple test results for system' do
```
Comment (skateman) [inline spec/controllers/v2/test_results_controller_spec.rb:119]: Why is this needed?

File: app/models/v2/test_result.rb [hunk 2]
```diff
@@ -118,6 +118,10 @@ def security_guide_version
       attributes['security_guide__version'] || try(:security_guide)&.version
     end
 
+    def last_scanned
+      system.test_results.max_by(&:end_time).end_time
```
Comment (skateman) [inline app/models/v2/test_result.rb:122]: N+1

File: app/serializers/v2/test_result_serializer.rb [hunk 3]
```diff
@@ -13,5 +13,6 @@ class TestResultSerializer < V2::ApplicationSerializer
     derived_attribute :compliant, :score, report: [:compliance_threshold]
     derived_attribute :system_id, :system_id
     derived_attribute :security_guide_version, security_guide: [:version]
+    derived_attribute :last_scanned, :end_time
```
Comment (skateman) [inline app/serializers/v2/test_result_serializer.rb:16]: What about other timestamp attributes? Like `:stale_timestamp, :stale_warning_timestamp, :updated, culled_timestamp`?

**Review by skateman — CHANGES_REQUESTED** [review]
OpenAPI spec changes?

File: app/models/v2/test_result.rb (hunk 2 above)
Comment (skateman) [reply app/models/v2/test_result.rb:122]: Also isn't this an own field? This does not even make sense.

Comment (romanblanco) [reply app/models/v2/test_result.rb:122]: According to the [schema description](https://github.com/RedHatInsights/compliance-backend/blob/master/spec/api/v2/schemas/test_result.rb#L116), the `last_scanned` returns _the date when the System has been reported a Test Result for the last time._

This lists `test_results` for the assigned systems and takes the latest `end_time`. What does not make sense to you?

Comment (skateman) [reply app/models/v2/test_result.rb:122]: How are the two things different in the context of the test result model? Also it's a very inefficient N+1 if you try to list test results. Just check how many SQL queries a single API request to list all test results of a report does with this change.

**Review by skateman — APPROVED** [review]
The Seal of Approval
